src/prefect/futures.py

```diff
@@ -233,6 +235,60 @@ def __eq__(self, other):
         return self.task_run_id == other.task_run_id
 
 
+class PrefectFutureList(list, Iterator, Generic[F]):
+    """
+    A list of Prefect futures.
+
+    This class provides methods to wait for all futures
+    in the list to complete and to retrieve the results of all task runs.
+    """
+
+    def wait_for_all(self, timeout: Optional[float] = None) -> None:
+        """
+        Wait for all futures in the list to complete.
+
+        Args:
+            timeout: The maximum number of seconds to wait for all futures to
+                complete. This method will not raise if the timeout is reached.
+        """
+        try:
+            with timeout_context(timeout):
+                for future in self:
+                    future.wait()
+        except TimeoutError:
+            logger.debug("Timed out waiting for all futures to complete.")
+            return
+
+    def results(
```

Comment:
I don't think it's necessary to pluralize this - the result is a list, and that's fine

Reply:
Makes sense. I also changed `wait_for_all` to `wait` so you can use the same method for one future or a list of futures.